docs/configuration/overview/runtime.rst

```diff
@@ -33,7 +33,51 @@ override_subdirectory
   specified and the directory exists, configuration values within this directory will override those
   found in the primary subdirectory. This is useful when Envoy is deployed across many different
   types of servers. Sometimes it is useful to have a per service cluster directory for runtime
-  configuration.
+  configuration. See below for exactly how the override directory is used.
+
+File system layout
+------------------
+
+Various sections of the configuration guide describe the runtime settings that are available.
+For example, :ref:`here <config_cluster_manager_cluster_runtime>` are the runtime settings for
+upstream clusters.
+
+Assume the following configuration settings for runtime:
+
+* *symlink_root*: ``/srv/runtime/current``
+* *subdirectory*: ``envoy``
+* *override_subdirectory*: ```envoy_override``
+
+Each '.' in a runtime key indicates a new directory in the hierarchy, rooted at *symlink_root* +
+*subdirectory*. For example, the *health_check.min_interval* path would have the following full
+file system path:
+
+``/srv/runtime/current/envoy/health_check/min_interval``
+
+The terminal portion of the path is a file that generally contains a numeric value unless otherwise
+documented. Envoy will strip white space as needed when reading the file.
+
+The *override_subdirectory* is used along with the :option:`--service-cluster` CLI option. Assume
+that :option:`--service-cluster` has been set to ``my-clustser``. Envoy will first look for the
+*health_check.min_interval* key in the following full file system path:
+
+``/srv/runtime/current/envoy_override/my-cluster/health_check/min_interval``
+
+If found, the value will override any value found in the primary lookup path. This allows runtime to
+be customized for individual clusters as needed on top of a global base.
+
+Symbolic link management
+------------------------
+
+Envoy uses a fairly common pattern for atomic file system updates. Envoy assumes that an entire
```

Comment:
There are two steps to update any runtime value. First, create a hard copy of the entire runtime tree and update the desired runtime values. Second, atomically swap the symlink root from the old tree to the new runtime tree, using the equivalent of the following command: